client/src/components/Input.js

```diff
@@ -1,13 +1,31 @@
 import styled from '@emotion/styled';
 import colors from '../utils/colorsUtils';
 
+const calculateWidth = (props) => {
+  if (props.Login) {
+    return '250px';
+  } else if (props.Search) {
```

Comment:
A react property should not be in PascalCase.
If you add custom props, use camelCase like the other existing props.

Another issue is, that you created a gernic Input component, which is great. But this component is not so generic anymore because it has special cases if it is used in login or in the search. 

Think about small components you can use for other projects too. 
I recommend to remove the props and export the Input with a default height and width. 
If you want to modify the default input, you can style it like this:

```
const LoginInput = styled(Input)`
  height: 50px.
`;
```

But I don't think you should set the height and width at all. It should be calculated automatically by the content and font-size.
If you have a button designed, take a look at the padding and font-size. This is all you need.
If you set a fixed width, the button would have issues with long content.

For input fields the form or container should set the width.

Reply:
First of all, thank you for your feedback!
I don't understand, why I should build and import an Input component, with basically default values, instead of building a custom Input, using the default input as a basis.
And why would I not set width and height for my components, if I want them all to have the same width and height? If the width of a button, for example, is calculated by the content and the font-size, every button has a different size, which is not what I wanted in the first place.
It is possible, that I misunderstand, what you are trying to tell me, in this case, I'm sorry!